pkg/controllers/state/cluster.go

```diff
@@ -82,6 +83,17 @@ func NewCluster(clk clock.Clock, client client.Client) *Cluster {
 //
 //nolint:gocyclo
 func (c *Cluster) Synced(ctx context.Context) (synced bool) {
+	// Set the metric depending on the result of the Synced() call
+	defer func() {
+		if c.unsyncedStartTime.IsZero() {
+			c.unsyncedStartTime = c.clock.Now()
+		}
+		if synced {
+			ClusterStateUnsyncedTimeSeconds.With(map[string]string{}).Set(0)
```

Comment:
This doesn't reset when the value gets synced. I think you need to set `c.unsyncedStartTime` to zero whenever you go synced

Reply:
I misunderstood what we wanted to set to 0 earlier. Fixed this.